channels/channel_state.go

```diff
@@ -171,6 +173,14 @@ func (c channelState) OtherPeer() peer.ID {
 	return c.sender
 }
 
+func (c channelState) Stages() *datatransfer.ChannelStages {
+	if c.stages == nil {
+		return &datatransfer.ChannelStages{}
```

Comment:
This behaviour a bit footgunny. This method returns a mutable `ChannelStages` object, either from the state (which means that changes will be persisted), or an entirely new instance without setting it in the internal state (which means that mutations will be discarded).

Reply:
It looks like we did it this way to save nil-checks on the user side, especially for transfers that were already in progress before this code got deployed. I'll come up with a different way.

Reply:
I ended up adding a note to the godocs that mutating the return value is unsafe. There's only so much you can do with Go. Even if we returned a value instead of a pointer, the inner slices can be mutated.